pages/api/users/[id].js

```diff
@@ -0,0 +1,20 @@
+import connect from "@/db/connect";
+import User from "@/db/models/User";
+
+export default async function handler(request, response) {
+  await connect();
+  if (request.method === "GET") {
+    const user = await User.findById(request.query.id)
+      .populate("listings")
+      .populate("starships")
+      .exec();
+    if (!user) {
+      response.status(404).json({
+        message:
+          "Sorry, something went wrong, 404 page not found! Please try again later.",
```

Comment:
The phrase "404 page not found" reads like the error page isn't being found. Maybe just leave this phrase out as the "404" is more for the devs than the users.